include/ignition/rendering/base/BaseCamera.hh

```diff
@@ -522,6 +578,37 @@ namespace ignition
       return result;
     }
 
+    //////////////////////////////////////////////////
+    template <class T>
+    void BaseCamera<T>::SetProjectionType(CameraProjectionType _type)
+    {
+      this->projectionType = _type;
+    }
+
+    //////////////////////////////////////////////////
+    template <class T>
+    CameraProjectionType BaseCamera<T>::ProjectionType() const
+    {
+      return this->projectionType;
+    }
+
+    //////////////////////////////////////////////////
+    template <class T>
+    math::Vector2i BaseCamera<T>::Project(const math::Vector3d &_pt)
+    {
+      math::Vector2i screenPos;
+      math::Matrix4d m = this->ProjectionMatrix() *  this->ViewMatrix();
+      math::Vector3d pos =  m * _pt;
+      double w = m(3, 0) * _pt.X() + m(3, 1) * _pt.Y() + m(3, 2) * _pt.Z()
+          + m(3, 3);
+      pos.X() = pos.X() / w;
+      pos.Y() = pos.Y() / w;
+
+      screenPos.X() = ((pos.X() / 2.0) + 0.5) * this->ImageWidth();
+      screenPos.Y() = (1 - ((pos.Y() / 2.0) + 0.5)) * this->ImageHeight();
```

Comment:
There are new Windows warnings:

```
'=': conversion from 'T' to 'T', possible loss of data
```

Reply:
testing fix in 6e2e25f

Reply:
windows build is now green as of 7a4ffa7